test/torchaudio_unittest/common_utils/case_utils.py

```diff
@@ -94,32 +95,107 @@ class TorchaudioTestCase(TestBaseMixin, PytorchTestCase):
+            # whole class is decorated
+            def _f(self, *_args, **_kwargs):
+                raise RuntimeError(reason)
+
+            test_item.setUp = _f
+            return test_item
+
+        # A method is decorated
+        @functools.wraps(test_item)
+        def f(*_args, **_kwargs):
+            raise RuntimeError(reason)
+
+        return f
+
+    return deco
+
+
+def _pass(test_item):
+    return test_item
+
+
+_IN_CI = _eval_env("CI", default=False)
+
+
+def _skipIf(condition, reason, key):
+    if not condition:
+        return _pass
+
+    # In CI, default to fail, so as to prevent accidental skip.
+    # In other env, default to skip
+    var = f"TORCHAUDIO_TEST_ALLOW_SKIP_IF_{key}"
+    skip_allowed = _eval_env(var, default=False if _IN_CI else True)
+    if skip_allowed:
+        return unittest.skip(reason)
+    return _fail(f"{reason} But the test cannot be skipped. (CI={_IN_CI}, {var}={skip_allowed}.)")
+
+
 def skipIfNoExec(cmd):
-    return unittest.skipIf(shutil.which(cmd) is None, f"`{cmd}` is not available")
+    return _skipIf(
+        shutil.which(cmd) is None,
+        f"`{cmd}` is not available",
+        key=f"NO_CMD_{cmd.upper().replace('-', '_')}",
+    )
 
 
 def skipIfNoModule(module, display_name=None):
-    display_name = display_name or module
-    return unittest.skipIf(not is_module_available(module), f'"{display_name}" is not available')
-
-
-def skipIfNoCuda(test_item):
-    if torch.cuda.is_available():
-        return test_item
-    force_cuda_test = os.environ.get("TORCHAUDIO_TEST_FORCE_CUDA", "0")
-    if force_cuda_test not in ["0", "1"]:
-        raise ValueError('"TORCHAUDIO_TEST_FORCE_CUDA" must be either "0" or "1".')
-    if force_cuda_test == "1":
-        raise RuntimeError('"TORCHAUDIO_TEST_FORCE_CUDA" is set but CUDA is not available.')
-    return unittest.skip("CUDA is not available.")(test_item)
-
-
-skipIfNoSox = unittest.skipIf(not is_sox_available(), reason="Sox not available")
-skipIfNoKaldi = unittest.skipIf(not is_kaldi_available(), reason="Kaldi not available")
-skipIfNoCtcDecoder = unittest.skipIf(not is_ctc_decoder_available(), reason="CTC decoder not available")
-skipIfRocm = unittest.skipIf(
-    os.getenv("TORCHAUDIO_TEST_WITH_ROCM", "0") == "1", reason="test doesn't currently work on the ROCm stack"
+    return _skipIf(
+        not is_module_available(module),
+        f'"{display_name or module}" is not available',
+        key=f"NO_MOD_{module.replace('.', '_')}",
+    )
+
+
+skipIfNoCuda = _skipIf(
+    not torch.cuda.is_available(),
+    reason="CUDA is not available.",
+    key="NO_CUDA",
+)
+skipIfNoSox = _skipIf(
+    not is_sox_available(),
+    reason="Sox features are not available.",
+    key="NO_SOX",
+)
+skipIfNoKaldi = _skipIf(
+    not is_kaldi_available(),
+    reason="Kaldi features are not available.",
+    key="NO_KALDI",
+)
+skipIfNoCtcDecoder = _skipIf(
+    not is_ctc_decoder_available(),
+    reason="CTC decoder not available",
```

Comment:
nit — so that the message passed to `_fail` is more grammatically correct

```suggestion
    reason="CTC decoder not available.",
```
similar for the rocm case below